cuvarbase/lombscargle.py

```diff
@@ -1080,16 +1082,44 @@ def batched_run_const_nfreq(self, data, batch_size=10,
         [mem.allocate(nf=nf, **kwargs) for mem in memory]
 
         funcs = (self.function_tuple, self.nfft_proc.function_tuple)
+        periods_best, significances = [], []
+
+        if doRemoveTerrestrial:
+            idx1 = np.where((freqs < 1.95) | (freqs > 2.05))[0]
+            idx2 = np.where((freqs < 0.95) | (freqs > 1.05))[0]
+            idx3 = np.where((freqs < 0.48) | (freqs > 0.52))[0]
+            idxterr = np.intersect1d(np.intersect1d(idx1,idx2),idx3)
+
         for b, batch in enumerate(batches):
+ 
             results = self.run(batch, memory=memory, freqs=freqs,
                                use_fft=use_fft,
                                **kwargs)
             self.finish()
-
+            
             for i, (f, p) in enumerate(results):
-                lsps.append(np.copy(p))
-
-        return [(freqs, lsp) for lsp in lsps]
+                if returnBestFreq:
+                    if doRemoveTerrestrial:
+                        powers = np.copy(p)[idxterr]
+                    else:
+                        powers = np.copy(p)
+
+                    fap = fap_baluev(batch[i][0], batch[i][2], powers, np.max(freqs))
+                    idx = np.argmin(fap)
+                    significance = 1./fap[idx]
```

Comment:
right now the FAP measure from Baluev 2008 is a raw false alarm probability. It doesn't take too much for this to saturate at either 1 or 0 (mainly 0), so doing 1/FAP is just a recipe for `inf`. What I'd like to do at some point is derive and write a `negative_fap_logit_baluev` function, which will return the negative *logit* of the false alarm probability. This would be much more numerically stable (-infinity for FAP=0, 0 for FAP=0.5, and +infinity for FAP = 1, but infinities will be much rarer).

until then i would keep this as `significance = 1 - fap[idx]`.